packages/windbg.vm/tools/chocolateyinstall.ps1

```diff
@@ -5,17 +5,32 @@ try {
     $toolName = 'WinDbg'
     $category = 'Debuggers'
 
-    # It seems WinDbg is now distributed as an .appinstaller and we need to install it using Add-AppxPackage
-    Add-AppxPackage -AppInstallerFile 'https://windbg.download.prss.microsoft.com/dbazure/prod/1-0-0/windbg.appinstaller'
+    $bundleUrl = "https://windbg.download.prss.microsoft.com/dbazure/prod/1-2402-24001-0/windbg.msixbundle"
+    $bundleSha256 = "e941076cb4d7912d32a22ea87ad2693c01fa465227b4d1ead588283518de428f"
 
-    $shortcutDir = Join-Path ${Env:TOOL_LIST_DIR} $category
-    $shortcut = Join-Path $shortcutDir "$toolName.lnk"
-    $executableCmd  = Join-Path ${Env:WinDir} "system32\cmd.exe"
-    # Use `start` to close the open console
-    $executableArgs = "/C start WinDbgX.exe"
-    $executableDir  = Join-Path ${Env:UserProfile} "Desktop"
-    Install-ChocolateyShortcut -shortcutFilePath $shortcut -targetPath $executableCmd -Arguments $executableArgs -WorkingDirectory $executableDir -RunAsAdmin
+    $packageArgs = @{
+        packageName   = ${Env:ChocolateyPackageName}
+        url           = $bundleUrl
+        checksum      = $bundleSha256
+        checksumType  = "sha256"
+        fileFullPath  = Join-Path ${Env:TEMP} "$toolName.msixbundle"
+    }
+    Get-ChocolateyWebFile @packageArgs
+    Add-AppxPackage -Path $packageArgs.fileFullPath
+
+    $installDir = (Get-AppxPackage -Name "Microsoft.$toolName").InstallLocation
+    $iconLocation  = Join-Path $installDir "DbgX.Shell.exe" -Resolve
+    $executablePath = "$(where.exe WinDbgXA.exe)"
```

Comment:
Actually `where.exe` doesn't throw exceptions, it returns `1` in case of failure and since `$ErrorActionPreference` is set to `Stop`; the non-zero return code is treated as an exception with the `stderr` as exception message.

It's a way better to use the powershell function `Get-Command` with the `ErrorAction` set to `Continue` and to check the returned path if empty or not instead of relying on exceptions.

It would be like this:
```powershell
$executablePath = (Get-Command WinDbgXA.exe -ea 0).Path
if (!$executablePath)
{
    $executablePath = Join-Path $installDir DbgX.Shell.exe
}
```

Reply:
@d35ha your proposal could cause that `$executablePath` is set to `Join-Path $installDir DbgX.Shell.exe` which doesn't fail the package, but we know fails to open in Windows 10 because of being in a protected directory. This results in a broken shortcut without noticing during the installation, or am I missing something?